### DISCUSSION TITLE
Why does openutau try to use a bat instead of directly using the resampler/wavtool on mac ?

### DISCUSSION THREAD

**Discussion by SeleDreams** [stakira/OpenUtau#719] [General]
I've been working on porting openutau to mac m1 since the current available binary doesn't seem to actually be working, however I am right now having an issue where it tries to load a bat file when starting open source resamplers and wavtools i built (since worldline is not available for M1 macs and is not open source)
because of it it just crashes with a bad exec error since bat files don't work on mac
obviously i most likely could fix it by making it use a script adapted to mac OS, but first of all I try to understand why it is working like this to begin with. is there a specific reason it has to rely on a bat file ?

**Comment by SeleDreams** [comment]
checking more the bat, it does look autogenerated, but i can't find exactly where it gets generated

**Comment by oxygen-dioxide** [comment]
> I've been working on porting openutau to mac m1 since the current available binary doesn't seem to actually be working, however I am right now having an issue where it tries to load a bat file when starting open source resamplers and wavtools i built (since worldline is not available for M1 macs and is not open source) because of it it just crashes with a bad exec error since bat files don't work on mac obviously i most likely could fix it by making it use a script adapted to mac OS, but first of all I try to understand why it is working like this to begin with. is there a specific reason it has to rely on a bat file ?

The bat is generated at OpenUtau.Core/Classic/ExeWavtool.cs https://github.com/stakira/OpenUtau/blob/b0e41dcdf32696cac92eb41171c3bf2f5f1346a2/OpenUtau.Core/Classic/ExeWavtool.cs#L32 . OpenUtau use .bat to run exe wavtools because it has to be compatible with moresampler's wavtool which needs to get required informations from the .bat file.
